src/Common/Services/EnvironmentDeploymentHandler.cs

```diff
@@ -78,6 +106,75 @@ private async Task ProcessEnvironmentDeployment(CustomerEnvironment environment,
             }
         }
 
+        private async Task UpdateQueuePositionAsync(long deployableId, CancellationToken token)
+        {
+            bool presentingLoading = false;
+            int? position = null;
+            int initialTopLine;
+            string msg;
+
+            while (!this._isDeploymentFinished && !_hasDeploymentStarted && !token.IsCancellationRequested)

```

Comment:
same in this condition. Do we need both this._isDeploymentFinished and !_hasDeploymentStarted?